Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java

```diff
@@ -1017,18 +1018,34 @@ public void searchForMedia(Process process, Workpiece workpiece) throws InvalidI
         }
         List<String> canonicals = getCanonicalFileNamePartsAndSanitizeAbsoluteURIs(workpiece, subfolders,
             process.getProcessBaseUri());
-        addNewURIsToExistingMediaUnits(mediaToAdd, workpiece.getAllMediaUnitsSorted(), canonicals);
+        addNewURIsToExistingMediaUnits(mediaToAdd, workpiece.getAllMediaUnitsFilteredByTypePageAndSorted(), canonicals);
         mediaToAdd.keySet().removeAll(canonicals);
         addNewMediaToWorkpiece(canonicals, mediaToAdd, workpiece);
         renumberMediaUnits(workpiece, true);
         if (ConfigCore.getBooleanParameter(ParameterCore.WITH_AUTOMATIC_PAGINATION)) {
             repaginateMediaUnits(workpiece);
         }
+        if (Workpiece.treeStream(workpiece.getRootElement())
+                .allMatch(includedStructuralElement -> includedStructuralElement.getViews().isEmpty())) {
+            automaticallyAssignMediaUnitsToEffectiveRootRecursive(workpiece, workpiece.getRootElement());
+        }
         if (logger.isTraceEnabled()) {
             logger.trace("Searching for media took {} ms", TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - begin));
         }
     }
 
+    private void automaticallyAssignMediaUnitsToEffectiveRootRecursive(Workpiece workpiece,
```

Comment:
why "automatically" ? Please remove this prefix

Reply:
Automatically, because you no longer have to manually assign it to the included structural element using drag and drop.